wezterm-ssh/src/config.rs

```diff
@@ -5,6 +5,20 @@ use std::path::{Path, PathBuf};
 
 pub type ConfigMap = BTreeMap<String, String>;
 
+pub trait ConfigMapOps {
+    fn is_yes(&self, name: &'static str) -> bool;
+    fn as_str(&self, name: &'static str) -> Option<&str>;
+}
+
+impl ConfigMapOps for ConfigMap {
+    fn is_yes(&self, name: &'static str) -> bool {
+        Some("yes") == self.get(name).map(|s| s.as_str())
+    }
+    fn as_str(&self, name: &'static str) -> Option<&str> {
+        self.get(name).map(|s| s.as_str())
+    }
+}
+
```

Comment:
Heh, this is (un?)surprisingly the most contentious part of this PR.  I have opinions; I think they are minor but there are several!

First, I think the use of a trait here is a walking a line of laziness; I say that without judgement, as economy of line-of-code count is generally desirable. I think that it makes things a bit repetitive here, which feels like more effort overall. It's not so bad that I hate it, but it doesn't quite sit with me.

What I propose instead is to make `ConfigMap` a newtype wrapper that derefs to the underlying map, so it "feels like" inheritance. Yes, it is more lines of code, but less repetition:

```rust
pub struct ConfigMap(BTreeMap<String, String>);

impl std::ops::Deref for ConfigMap {
   type Target = BTreeMap<String, String>;
   fn deref(&self) -> &Self::Target {
      &self.0
   }
}

impl std::ops::DerefMut for ConfigMap {
   fn deref_mut(&mut self) -> &mut Self::Target {
     &mut self.0
   }
}
```

Then these helper methods can be added as methods more directly; I suggest slightly different naming and implementation, something like this:

```rust
impl ConfigMap {
   pub fn get_str(&self, key: &str) -> Option<&str> {
      self.get(key).as_deref()
   }

   pub fn is(&self, key: &str, candidate: &str) -> bool {
      self.get_str(key) == Some(candidate)
   }

   pub fn is_yes(&self, key: &str) -> bool {
       self.is(key, "yes")
   }
}
```

Reply:
Okay,  I gave it a try.  Had to add a ConfigMap::new, and then was still left with errors
for callers not being able to use it as an iterator.  Making it an iterator is beyond my
knowledge at this point.  Sorry.

But this really isn't an integral part of the feature request; i'd suggest we just go
with the first 3 commits, and drop this one altogether for now.  Can always circle
back later.

Reply:
re: iterator, there are some iterator traits but I think declaring them is probably not warranted for this. You should be able to explicitly call `.iter()` on the newtype to get at the iterator. 

I'm happy to take the first three commits for now.

Reply:
Okay, dropped that part of the series, and pushed.